lib/src/method_call_handler.dart

```diff
@@ -9,17 +9,32 @@ class AblyMethodCallHandler {
       switch (call.method) {
         case PlatformMethod.authCallback:
           return await onAuthCallback(call.arguments);
+        case PlatformMethod.realtimeAuthCallback:
+          return await onRealtimeAuthCallback(call.arguments);
       }
     });
   }
 
-  Future<dynamic> onAuthCallback(AblyMessage message) async {
-    ably.TokenParams tokenParams = message.message as ably.TokenParams;
-    ably.Rest rest = ably.restInstances[message.handle];
+  Future<Object> onAuthCallback(AblyMessage message) async {
+    var tokenParams = message.message as ably.TokenParams;
+    var rest = ably.restInstances[message.handle];
     final callbackResponse = await rest.options.authCallback(tokenParams);
-    Future.delayed(Duration.zero, () {
-      rest.channels.all.forEach((c) => c.authUpdateComplete());
-    });
+    Future.delayed(Duration.zero, rest.authUpdateComplete);
+    return callbackResponse;
+  }
+
+  bool realtimeAuthInProgress = false;
```

Comment:
I missed that this should be private

Reply:
Addressed 0ab957b7fed08750fc0893be45b0f5a0f13878af